core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -111,6 +110,14 @@ class SparkHadoopUtil extends Logging {
   }
 
   /**
+    * Return an appropriate (subclass) of Configuration. Creating config can initializes some Hadoop
+    * subsystems.
+    */
+  def newConfiguration(conf: SparkConf): Configuration = {
```

Comment:
This is being used from a bunch of other places in the code base. I agree with you that its not a good idea to create a new Configuration instead of reusing it and we should change that behavior.